xdsl/dialects/riscv.py

```diff
@@ -159,6 +161,31 @@ class RISCVOp(Operation, ABC):
     pass
 
 
+class RISCVPrinterInterface(ABC):
```

Comment:
Why not have this on the RISCVOp directly?

I would suggest a superclass abstract method with the following implementation:
``` python
@abstractmethod
def riscv_assembly_line(self) -> str | None:
    raise NotImplemented
```

And then a subclass that the majority of ops can implement:
``` python
class RISCVInstruction(RISCVOp):

    def riscv_assembly_line(self) -> str | None:
        ... # implementation from riscv_asm_writer
    @abstractmethod
    def riscv_printed_name(self) -> str:
        """
        Give the name of the RISC-V instruction
        """
        raise NotImplementedError()

    @abstractmethod
    def riscv_printed_components(
        self,
    ) -> Sequence[
        IntegerAttr[IntegerType | IndexType] | LabelAttr | SSAValue | str | None
    ]:
        """
        Return the list of "arguments" to the operation
        """
        raise NotImplementedError()
```